client/guidestours/index.js

```diff
@@ -27,8 +29,18 @@ const posToCss = ( { x, y } ) => ( {
 const middle = ( a, b ) => Math.abs( b - a ) / 2;
 
 const dialogPositioners = {
-	below: ( { left, bottom } ) => ( { x: left + DIALOG_PADDING, y: bottom + DIALOG_PADDING } ),
-	beside: ( { right, top } ) => ( { x: right + DIALOG_PADDING, y: top + DIALOG_PADDING } ),
+	below: ( { left, bottom } ) => ( {
+		x: ( left + DIALOG_PADDING + DIALOG_WIDTH ) > document.documentElement.clientWidth
+			? ( document.documentElement.clientWidth - DIALOG_PADDING - DIALOG_WIDTH )
+			: ( left + DIALOG_PADDING ),
+		y: bottom + DIALOG_PADDING
+	} ),
+	beside: ( { left, right, top } ) => ( {
+		x: ( right + DIALOG_PADDING + DIALOG_WIDTH ) > document.documentElement.clientWidth
+			? ( Math.abs( left - DIALOG_PADDING - DIALOG_WIDTH ) )
+			: ( right + DIALOG_PADDING ),
+		y: top + DIALOG_PADDING
+	} ),
```

Comment:
These are now pretty hard to read. Can we improve the semantics by hiding some bits behind some functions?